osu.Game.Rulesets.Catch/Edit/Blueprints/JuiceStreamSelectionBlueprint.cs

```diff
@@ -168,6 +190,48 @@ private void updateHitObjectFromPath()
             lastSliderPathVersion = HitObject.Path.Version.Value;
         }
 
+        private void convertToStream()
```

Comment:
Not sure how to feel about this being a nigh carbon-copy of the osu! implementation. Especially so that I've heard osu! mappers call the current implementation of slider-to-stream "kneecapped" and wish for the stable one that could do more to be restored.

Probably would have preferred this was split to some common method in light of that, so that once that gets implemented, both rulesets reap the benefit. Or at the minimum link back both copies of the logic to each other.

Reply:
> Not sure how to feel about this being a nigh carbon-copy of the osu! implementation. Especially so that I've heard osu! mappers call the current implementation of slider-to-stream "kneecapped" and wish for the stable one that could do more to be restored.

I wasn't aware that the current implementation was subpar; My intention for this PR was to have something in place instead of nothing at all. So might be better to have an improved version of this (and the osu version improved) in a follow-up PR.

> Probably would have preferred this was split to some common method in light of that, so that once that gets implemented, both rulesets reap the benefit. Or at the minimum link back both copies of the logic to each other.

Good point. I'll rework it a bit so we have a shared implementation for the duplicate logic 👍
I initially didn't want to do this as I wasn't sure how overlapping the catch and osu logic would be, but it ended up almost the same...

Reply:
Given how small the method is I think it's fine to just go with what we have for now to get the functionality in?

Reply:
Added comment backreferences in https://github.com/ppy/osu/pull/30763/commits/5a0b732ee32e66e24118e390706795a419cd3954. Can be cleaned up whenever convenient.